src/ui/components/Passport/Passport.tsx

```diff
@@ -283,50 +283,29 @@ const Passport = () => {
   }, [selectedItem.completed]);
 
   return (
-    <div className={styles.PassportBoxWrapper}>
-      <div className={styles.PassportBoxGradient}>
-        <div className={styles.PassportBox}>
-          <div className="p-0 pt-3">
-            <img src={`/images/passport/passportHeader.svg`} className={`mb-5 w-full px-1`} />
-            <div
-              className={styles.PassportBoxInternal}
-              style={{
-                height: videoHeight !== null ? `calc(100vh - 200px - ${videoHeight}px)` : "100%",
-              }}
-            >
-              <div className={styles.sectionsContainer}>
-                {sections.map((section, sectionIndex) => (
-                  <React.Fragment key={sectionIndex}>
-                    {renderSection(section, sectionIndex)}
-                  </React.Fragment>
-                ))}
-              </div>
-            </div>
-          </div>
-        </div>
-        {selectedItem.completed && (
-          <img
-            src={`/images/passport/${selectedItem.imageBaseName}-complete.png`}
-            className={styles.largeCompletedImage}
-            style={{
-              zIndex: 0,
-              opacity: 0.25,
-              bottom: `${randomBottom}px`,
-              right: `${randomRight}px`,
-              transform: `rotate(${randomRotation}deg)`,
-            }}
-          />
-        )}
-        <div className={styles.videoExampleWrapper}>
-          <div style={{ position: "relative" }}>
-            <img
-              src={selectedItem.videoUrl}
-              className={styles.videoExample}
-              ref={videoExampleRef}
-            />
+    <div className={`${styles.PassportBoxWrapper} flex h-screen flex-col`}>
+      <div className="my-2 p-2">
+        <img src={`/images/passport/passportHeader.svg`} className={`w-full px-1`} />
+      </div>
+      <div className="flex-grow overflow-auto">
+        <div className="p-2">
+          <div className={styles.sectionsContainer}>
+            {sections.map((section, sectionIndex) => (
+              <React.Fragment key={sectionIndex}>
+                {renderSection(section, sectionIndex)}
+              </React.Fragment>
+            ))}
           </div>
         </div>
       </div>
+
+      <div className="p-1">
+        <img
+          src={selectedItem.videoUrl}
+          className={`${styles.videoExample} w-full object-cover`}
+          ref={videoExampleRef}
```

Comment:
I found myself curious why you were using a ref here.

Looks like there are duplicate effects:
https://github.com/replayio/devtools/blob/c134c0cdbb8359fdd2526ae1ca10472f6b7bdc72/src/ui/components/Passport/Passport.tsx#L47-L68

You could consolidate those:
```js
useLayoutEffect(() => {
  const videoExample = videoExampleRef.current;
  if (videoExample) {
    const updateHeight = () => {
      setVideoHeight(videoExample.offsetHeight);
    };

    window.addEventListener("resize", updateHeight);

    return () => {
      window.removeEventListener("resize", updateHeight);
    };
  }
}, []);
```

So...
```diff
diff --git a/src/ui/components/Passport/Passport.tsx b/src/ui/components/Passport/Passport.tsx
index 59c8e3130..5e17b6387 100644
--- a/src/ui/components/Passport/Passport.tsx
+++ b/src/ui/components/Passport/Passport.tsx
@@ -44,27 +44,19 @@ const Passport = () => {
   const videoExampleRef = useRef<HTMLImageElement>(null);
   const [videoHeight, setVideoHeight] = useState<number | null>(null);
 
-  useEffect(() => {
-    if (videoExampleRef.current) {
-      const videoHeight = videoExampleRef.current.offsetHeight;
-      setVideoHeight(videoHeight);
-    }
-  }, [videoExampleRef]);
+  useLayoutEffect(() => {
+    const videoExample = videoExampleRef.current;
+    if (videoExample) {
+      const updateHeight = () => {
+        setVideoHeight(videoExample.offsetHeight);
+      };
 
-  useEffect(() => {
-    const updateHeight = () => {
-      if (videoExampleRef.current) {
-        const videoHeight = videoExampleRef.current.offsetHeight;
-        setVideoHeight(videoHeight);
-        console.log("Updating", videoHeight);
-      }
-    };
+      window.addEventListener("resize", updateHeight);
 
-    window.addEventListener("resize", updateHeight);
-
-    return () => {
-      window.removeEventListener("resize", updateHeight);
-    };
+      return () => {
+        window.removeEventListener("resize", updateHeight);
+      };
+    }
   }, []);
 
   const getItemStyle = (sectionIndex: number, itemIndex: number) => {
```

Reply:
Thank you! Yeah, I changed so many things I lost the plot.